src/frontend/cellprofiler/modules/measureimagequality.py

```diff
@@ -1084,421 +1044,65 @@ def post_run(self, workspace):
-        area_text, area_measurement = (
-            ("Volume", F_TOTAL_VOLUME) if volumetric else ("Area", F_TOTAL_AREA)
-        )
-
-        pixel_count = numpy.product(pixels.shape)
-        if pixel_count == 0:
-            pixel_sum = 0
-            pixel_mean = 0
-            pixel_std = 0
-            pixel_mad = 0
-            pixel_median = 0
-            pixel_min = 0
-            pixel_max = 0
-        else:
-            pixel_sum = numpy.sum(pixels)
-            pixel_mean = pixel_sum / float(pixel_count)
-            pixel_std = numpy.std(pixels)
-            pixel_median = numpy.median(pixels)
-            pixel_mad = numpy.median(numpy.abs(pixels - pixel_median))
-            pixel_min = numpy.min(pixels)
-            pixel_max = numpy.max(pixels)
-
-        m = workspace.measurements
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, area_measurement, image_name)), pixel_count
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_TOTAL_INTENSITY, image_name)), pixel_sum
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_MEAN_INTENSITY, image_name)), pixel_mean
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_MEDIAN_INTENSITY, image_name)), pixel_median
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_STD_INTENSITY, image_name)), pixel_std
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_MAD_INTENSITY, image_name)), pixel_mad
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_MAX_INTENSITY, image_name)), pixel_max
-        )
-        m.add_image_measurement(
-            "_".join((C_IMAGE_QUALITY, F_MIN_INTENSITY, image_name)), pixel_min
-        )
-
-        result = [
-            ["{} {}".format(image_name, feature_name), "{:.2f}".format(value)]
-            for feature_name, value in (
-                ("Total intensity", pixel_sum),
-                ("Mean intensity", pixel_mean),
-                ("Median intensity", pixel_median),
-                ("Std intensity", pixel_std),
-                ("MAD intensity", pixel_mad),
-                ("Min intensity", pixel_min),
-                ("Max intensity", pixel_max),
-                ("Total {}".format(area_text), pixel_count),
-            )
-        ]
-        return result
-
-    def calculate_power_spectrum(self, image_group, workspace):
-        result = []
-        for image_name in self.images_to_process(image_group, workspace):
+        if image_names is None:
+            logging.warning("[MeasureImageQuality]: No images to process in the image group")
+            return []
+        
+        scale_groups = [scale_group.scale.value for scale_group in image_group.scale_groups]
+        all_threshold_groups = self.get_all_threshold_groups(image_group)
+        
+        for image_name in image_names:
```

Comment:
This is a significant change in the code for this module. Previous version iterated over images for every feature. As a result, the same image would load for each feature. The new version optimizes this by loading the image once and uses the same pixel data for all features.